itest/lnd_funding_test.go

```diff
@@ -777,6 +778,17 @@ func testBatchChanFunding(ht *lntest.HarnessTest) {
 	ht.AssertTopologyChannelOpen(alice, chanPoint2)
 	ht.AssertTopologyChannelOpen(alice, chanPoint3)
 
+	// Check if the change type from the batch_open_channel funding is P2TR.
+	rawTx := ht.Miner.GetRawTransaction(txHash)
+	require.Len(ht, rawTx.MsgTx().TxOut, 4)
+
+	changeIndex := uint32(6) - (chanPoint1.OutputIndex +
```

Comment:
I am not sure if the change-output is always the last output though, maybe just find the change by excluding the amounts? Its maybe constant for this scenario but not generally speaking because its sorted I think in the btcwallet package because there the funding takes place. 

Could you explain your thought behind your formula ?

Reply:
This formula doesn't take the last output. Here is the explanation!

If you have M channels points and 1 change-output, then the indexes go from 0 to M.
We know the OutputIndex of every channels points, so we just need the missing int between 0 and M.

To do that, we know how to calculate the sum of the M consecutive int (aka if there is no missing int):

`Ideal_Sum= M*(M+1)/2`

In our case, we know the sum of all the channel output indexes, so, to find out the missing int we just need to do:

Ideal_Sum- Sum(Channels outpoint indexes)

In out case M=3 so you have this formula (Ideal_Sum=6)

With an example: if channel indexes are 0,1,3 then change outpout index is 6-(0+1+3)=2

Reply:
Thanks for your explanation, nice again something learned :). Not sure if this ideal sum thing is common knowledge but I would suggest a nice comment maybe ?

Reply:
no problem! :) will add it!

Reply:
nit: your comment describes the code I would have prefered something like this:

Suggestion:
 ```
//For calculating the change output index we use the formula for consecutive sum of integers n(n+1)/2. In our case n=3. All the channel point indices are known so we just calculate the difference to get the change output index`
```

Reply:
Pretty neat indeed.